### PR TITLE
feat(processors): verify tests using functions from the processor

### LINKED ISSUE #261: Validate all javascript functions in test actually exists in the processor

### PR DESCRIPTION
**Pull request by enudler**
closes #261

### CODE REVIEW

File: src/tests/helpers/testsVerifier.js [hunk 1]
```diff
@@ -1,21 +1,49 @@
...
+
+        if (!errorToThrow) {
+            let usedFunctionsWhichNotExists = usedFunctions.filter(uf => !processor.exported_functions.includes(uf));
+            usedFunctionsWhichNotExists = _.uniq(usedFunctionsWhichNotExists);
+            if (usedFunctionsWhichNotExists.length > 0) {
+                errorToThrow = new Error(`Functions: ${usedFunctionsWhichNotExists.join(', ')} does not exist in the processor file`);
+                errorToThrow.statusCode = 400;
+            }
+        }
+    } else if (usedFunctions.length > 0 && testBody.type === consts.TEST_TYPE_BASIC) {
```
Comment (NivLipetz) [inline src/tests/helpers/testsVerifier.js]:
```suggestion
    } else if (usedFunctions.length > 0) {
```

we should go into this `if` as long as `!test.processor_id` and `usedFunctions.lenght>0`

File: tests/integration-tests/tests/tests-test.js [hunk 2]
```diff
@@ -55,6 +57,27 @@ describe('the tests api', function() {
...
+                description: 'This is a description',
+                javascript: 'module.exports.func = 5'
+            };
+            const processorResponse = await processorsRequestSender.createProcessor(processor, validHeaders);
+            processorResponse.statusCode.should.eql(201);
+            const processorId = processorResponse.body.id;
+            let requestBody = Object.assign({ processor_id: processorId }, testWithFunctions);
+            const createTestResponse = await testsRequestSender.createTest(requestBody, validHeaders);
+            createTestResponse.statusCode.should.eql(400);
+            createTestResponse.body.message.should.eql('Functions: beforeScenario, afterScenario, afterResponse, beforeRequest does not exist in the processor file');
```
Comment (NivLipetz) [inline tests/integration-tests/tests/tests-test.js]: shouldn't this error be 
```suggestion
            createTestResponse.body.message.should.eql('Functions: beforeScenario, afterScenario, afterResponse, beforeRequest, abc does not exist in the processor file');
```

Comment (enudler) [reply tests/integration-tests/tests/tests-test.js]: you are right
we currently not supporting arrays also not in the ui
I will open a task to do it in the future

Comment (NivLipetz) [reply tests/integration-tests/tests/tests-test.js]: so lets update the test_with_functions file

**Pull request merged**